doc/tutorial.md

```diff
@@ -0,0 +1,607 @@
+}]] -- this is a multiline string
+
+
+return bd.system
+{
+   imports = {
+      "stdtypes",
+      "stattypes",
+      "ptrig",
+      "lfds_cyclic",
+      "logger",
+      "platform_2dof",
+      "platform_2dof_control",
+   },
+
+   blocks = {
+      { name="plat1", type="platform_2dof" },
+      { name="control1", type="platform_2dof_control" },
+      { name="logger_time", type="logging/file_logger" },
+      { name="fifo_pos", type="lfds_buffers/cyclic" },
+      { name="fifo_vel", type="lfds_buffers/cyclic" },
+      { name="ptrig1", type="std_triggers/ptrig" },
+   },
+   connections = {
+     { src="plat1.pos", tgt="fifo_pos" }, 
+     { src="fifo_pos",tgt="control1.measured_pos" },
+     { src="control1.commanded_vel",tgt="fifo_vel" },
+     { src="fifo_vel",  tgt="plat1.desired_vel" },
+
+   },
+   configurations = {
+      { name="plat1", config = {  initial_position={1.1,1}, joint_velocity_limits={0.5,0.5} } },
+      { name="control1", config = {  gain=0.1, target_pos={4.5,4.5} } },
+      { name="fifo_pos", config = { type_name="double", data_len=2, buffer_len=1 } },
+      { name="fifo_vel", config = { type_name="double", data_len=2, buffer_len=1 } },
+      { name="logger_time", config = { filename="/tmp/platform_time.log",
+                                          separator=",",
+                                          report_conf = timer_report_conf, } },
+      { name="ptrig1", config = { period = {sec=1, usec=0 },
+                                   sched_policy="SCHED_OTHER",
+                                   sched_priority=0,
+                                   trig_blocks={ { b="#plat1", num_steps=1, measure=1 },
+						 { b="#control1", num_steps=1, measure=1 },
+						 { b="#logger_time", num_steps=1, measure=0 }  } } }
+   }
+
+}
+```
+It is worth noticing that configuration types can be arrays (*e.g.* `target_pos`), strings (`file_name` and `report_conf`) and structures (`period`) and vector of structures (`trig_blocks`). Types for each porpoery should be checked in source files.
+Alternatively, the web server can provide insight of the types.
+
+The file is launched with the command 
+```bash
+ubx_ilaunch -c platform_2dof_and_control.usc
+```
+or 
+```bash
+ubx_ilaunch -webif -c platform_2dof_and_control.usc
+```
+to enable the webinterface at [localhost:8888](localhost:8888) .
+In order to visualise the data saved by the logger in the _\tmp_ folder, consider [kst](https://kst-plot.kde.org/) or any other program that can visualise a comma-separated-value file.
+###Some considerations about the fifos
+
+First of all, consider that each fifo can be connected with multiple input and pultiple oitput ports.
+Consider also, that if multiple out are connected, if one read one data, that data will be consumed and not available for a second port. In general cases, each fifo has a sinle input and a single output, while an output port can be connected with multiple output.
```

Comment:
This may be confusing to a new user. If you use the DSL, then each connection among ports will use it's own fifo and do what is expected in most cases. If you want to create a single output port -> multiple input ports connections, you need to specifically instantiate the iblocks and create the connections among them.